src/gurt/telemetry.c

```diff
@@ -1038,847 +1038,249 @@ d_tm_compute_stats(struct d_tm_node_t *node, uint64_t value)
  */
-int
-d_tm_mark_duration_start(struct d_tm_node_t **metric, int clk_id,
-			 const char *fmt, ...)
+void
+d_tm_mark_duration_start(struct d_tm_node_t *metric, int clk_id)
 {
-	struct d_tm_node_t	*node = NULL;
-	char			path[D_TM_MAX_NAME_LEN] = {};
-	int			rc = DER_SUCCESS;
-
-	if (d_tm_shmem_root == NULL)
-		return -DER_UNINIT;
-
-	if ((metric != NULL) && (*metric != NULL)) {
-		node = *metric;
-	} else {
-		va_list	args;
-		int	ret;
-
-		if (fmt == NULL) {
-			rc = -DER_INVAL;
-			goto out;
-		}
-
-		va_start(args, fmt);
-		ret = vsnprintf(path, sizeof(path), fmt, args);
-		va_end(args);
-
-		if (ret <= 0 || ret >= D_TM_MAX_NAME_LEN) {
-			rc = -DER_EXCEEDS_PATH_LEN;
-			goto out;
-		}
-
-		node = d_tm_find_metric(d_tm_shmem_root, path);
-		if (metric != NULL)
-			*metric = node;
-	}
-
-	if (node == NULL) {
-		if (!((clk_id == D_TM_CLOCK_REALTIME) ||
-		      (clk_id == D_TM_CLOCK_PROCESS_CPUTIME) ||
-		      (clk_id == D_TM_CLOCK_THREAD_CPUTIME))) {
-			rc = -DER_INVAL;
-			D_ERROR("Invalid clk_id for [%s] "
-				"Failed to add metric: " DF_RC "\n", path,
-				DP_RC(rc));
-			goto out;
-		}
-		rc = d_tm_add_metric(&node, D_TM_DURATION | clk_id,
-				     d_tm_clock_string(clk_id), NULL, path);
-		if (rc != DER_SUCCESS) {
-			D_ERROR("Failed to add and mark duration start [%s]: "
-				DF_RC "\n", path, DP_RC(rc));
-			goto out;
-		}
-		if (metric != NULL)
-			*metric = node;
-	}
+	if (d_tm_shmem_root == NULL || metric == NULL)
+		return;
 
-	if (!(node->dtn_type & D_TM_DURATION)) {
-		rc = -DER_OP_NOT_PERMITTED;
+	if (!(metric->dtn_type & D_TM_DURATION)) {
 		D_ERROR("Failed to mark duration start [%s] on item "
 			"not a duration.  Operation mismatch: " DF_RC "\n",
-			path, DP_RC(rc));
-		goto out;
+			metric->dtn_name, DP_RC(-DER_OP_NOT_PERMITTED));
+		return;
 	}
 
-	if (node->dtn_protect)
-		D_MUTEX_LOCK(&node->dtn_lock);
-	clock_gettime(d_tm_clock_id(node->dtn_type & ~D_TM_DURATION),
-		      &node->dtn_metric->dtm_data.tms[1]);
-	if (node->dtn_protect)
-		D_MUTEX_UNLOCK(&node->dtn_lock);
-
-out:
-	return rc;
+	d_tm_node_lock(metric);
+	clock_gettime(d_tm_clock_id(metric->dtn_type & ~D_TM_DURATION),
+		      &metric->dtn_metric->dtm_data.tms[1]);
+	d_tm_node_unlock(metric);
 }
 
 /**
  * Mark the end of the time interval started by d_tm_mark_duration_start()
  * Calculates the total interval and stores the result as the value of this
  * metric.
  *
- * The duration is specified either by an initialized pointer or by a fully
- * qualified item name.  If an initialized pointer is provided, the metric is
- * accessed directly.  Otherwise, a lookup is performed on the path name
- * provided in order to find the specified item.  Because this function must be
- * paired with d_tm_mark_duration_start(), the metric is not created if it
- * does not already exist.
- *
  * \param[in]		metric	Pointer to the metric
- * \param[in]		err	If non-zero, aborts the interval calculation
- *				so that this interval is not added to the stats.
```

Comment:
Was the removal of err done on purpose?  That was added at Johann's request not all that long ago.

Reply:
Yeah, I didn't see it used anywhere outside of tests, and it seemed like an odd pattern. If someone intends to use it, I'll happily add it back in.